app/bundles/LeadBundle/Model/ListModel.php

```diff
@@ -1195,4 +1200,73 @@ public function getStagesBarChartData($unit, \DateTime $dateFrom, \DateTime $dat
 
         return $chartData;
     }
+    /**
+     * Get bar chart data of hits
+     *
+     * @param char     $unit   {@link php.net/manual/en/function.date.php#refsect1-function.date-parameters}
+     * @param DateTime $dateFrom
+     * @param DateTime $dateTo
+     * @param string   $dateFormat
+     * @param array    $filter
+     *
+     * @return array
+     */
+    public function getDeviceGranularityData($unit, \DateTime $dateFrom, \DateTime $dateTo, $dateFormat = null, $filter = array())
+    {
+        $data['values'] = array();
+        $data['labels'] = array();
+
+        $q = $this->em->getConnection()->createQueryBuilder();
+
+        $q->select('count(l.id) as leads, h.device as device')
+            ->from(MAUTIC_TABLE_PREFIX.'lead_lists_leads', 't')
+            ->join('t', MAUTIC_TABLE_PREFIX.'leads', 'l', 'l.id = t.lead_id')
+            ->join('t', MAUTIC_TABLE_PREFIX.'page_hits','h', 'h.lead_id=l.id')
+            ->orderBy('device', 'DESC')
+            ->andWhere($q->expr()->gte('t.date_added', ':date_from'))
+            ->setParameter('date_from', $dateFrom->format('Y-m-d'))
+            ->andWhere($q->expr()->lte('t.date_added', ':date_to'))
+            ->setParameter('date_to', $dateTo->format('Y-m-d'." 23:59:59"));
+
+        if(isset($filter['leadlist_id']['value'])){
```

Comment:
There should be spaces after `if` and before `{`. This code standards issue is on more places with if and foreach statements.